### PR TITLE
use readline in obclient

### PR DESCRIPTION
**Pull request by hnwyllmm**
当系统安装readline时，obclient使用readline

### CODE REVIEW

File: src/obclient/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-MESSAGE(STATUS "This is PROJECT_BINARY_DIR dir " ${PROJECT_BINARY_DIR})
-MESSAGE(STATUS "This is PROJECT_SOURCE_DIR dir " ${PROJECT_SOURCE_DIR})
+ADD_EXECUTABLE(obclient)
+MESSAGE("Begin to build " obclient)
 
+INCLUDE(CheckIncludeFiles)
 
 #INCLUDE_DIRECTORIES([AFTER|BEFORE] [SYSTEM] dir1 dir2 ...)
-INCLUDE_DIRECTORIES(. ${PROJECT_SOURCE_DIR}/../../deps /usr/local/include SYSTEM)
+TARGET_INCLUDE_DIRECTORIES(obclient PRIVATE . ${PROJECT_SOURCE_DIR}/../deps /usr/local/include /usr/include)
```
Comment (longdafeng) [inline src/obclient/CMakeLists.txt:7]: 1. why not use ${PROJECT_SOURCE_DIR} replacing "object PRIVATE"?
2. why not use SYSTEM replacing "/usr/include"

File: src/obclient/CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
 # 父cmake 设置的include_directories 和link_directories并不传导到子cmake里面
 #INCLUDE_DIRECTORIES(BEFORE ${CMAKE_INSTALL_PREFIX}/include)
-LINK_DIRECTORIES(/usr/local/lib ${PROJECT_BINARY_DIR}/../../lib)
 
+TARGET_LINK_DIRECTORIES(obclient PRIVATE /usr/local/lib ${PROJECT_BINARY_DIR}/../../lib)
```
Comment (longdafeng) [inline src/obclient/CMakeLists.txt:11]: same as above

File: src/obclient/CMakeLists.txt [hunk 3]
```diff
@@ -19,15 +29,14 @@ FOREACH (F ${ALL_SRC})
...
 
-
 # 指定目标文件位置
-SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/../../bin)
+SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/../bin)
 MESSAGE("Binary directory:" ${EXECUTABLE_OUTPUT_PATH})
-ADD_EXECUTABLE(${PROJECT_NAME} ${PRJ_SRC})
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} common pthread dl)
+TARGET_SOURCES(obclient PRIVATE ${PRJ_SRC})
+TARGET_LINK_LIBRARIES(obclient common pthread dl)
```
Comment (longdafeng) [inline src/obclient/CMakeLists.txt:36]: same as above

File: src/obclient/CMakeLists.txt [hunk 4]
```diff
@@ -19,15 +29,14 @@ FOREACH (F ${ALL_SRC})
...
 
 
 # Target 必须在定义 ADD_EXECUTABLE 之后， programs 不受这个限制
 # TARGETS和PROGRAMS 的默认权限是OWNER_EXECUTE, GROUP_EXECUTE, 和WORLD_EXECUTE，即755权限， programs 都是处理脚步类
 # 类型分为RUNTIME／LIBRARY／ARCHIVE, prog
-INSTALL(TARGETS ${PROJECT_NAME}  RUNTIME DESTINATION bin)
\ No newline at end of file
+INSTALL(TARGETS obclient RUNTIME DESTINATION bin)
```
Comment (longdafeng) [inline src/obclient/CMakeLists.txt:42]: same as above

**Review by longdafeng — COMMENTED** [review]
Please take a look at comments

**Comment by hnwyllmm** [comment]
All subdirectoy can have the same view of the project, such as the source path, include directory, library path, and so on. 
Further more, this file can be simplified better and I can do it later.

**Pull request merged**